src/autotrain/nvcf.py

```diff
@@ -0,0 +1,35 @@
+
+import os
+from transformers import TrainerCallback, TrainerControl, TrainerState, TrainingArguments
+from accelerate import PartialState
+
+REQUEST_ID = os.getenv("NVCF-REQID")
+LOG_PAYLOAD = """
+    {{
+        \\"id\\": \\"{request_id}\\",
+        \\"progress\\": \\"{progress}\\"
+    }}
+"""
+
+LOG_PATH = "/var/inf/response/{request_id}/log.partialprogress"
+
+class NVCFCallback(TrainerCallback):
```

Comment:
dont like the callback way. then we need to add it to all other available tasks too